CONTRIBUTING.md

```diff
@@ -0,0 +1,91 @@
+# Contributing to itertools
+
+We use stable Rust only.
+Please check the minimum version of Rust we use in `Cargo.toml`.
+
+To pass CI tests, you should run `cargo fmt` and it might become mandatory that
+your code changes are free of "compile warnings" and "clippy warnings"
+(see [#720](https://github.com/rust-itertools/itertools/pull/720) and [#740](https://github.com/rust-itertools/itertools/pull/740)).
+
+No guideline about using `#[inline]` yet.
+
+## `core::iter::Iterator` vs `itertools::Itertools`
+For new features, please first consider filing a PR to [rust-lang/rust](https://github.com/rust-lang/rust),
+adding your new feature to the `Iterator` trait of the standard library, if you believe it is reasonable.
+If it isn't accepted there, proposing it for inclusion in `itertools` is a good idea.
+The reason for doing is this is so that we avoid future breakage as with `.flatten()`.
+However, if your feature involves heap allocation, such as storing elements in a `Vec<T>`,
+then it can't be accepted into `libcore`, and you should propose it for `itertools` directly instead.
+
```

Comment:
Let's start with: **File an issue proposing your feature.**

With some text to the effect of: *If you are proposing a major change to CI or a new iterator adaptor for this crate, please first file an issue describing your proposal.*

Reply:
I left a comment nearly at the beginning for it to be seen quickly.